### PR TITLE
Most of what's needed to wire up k8s info in logging for requests.

### PR DESCRIPTION
**Pull request by mikebell90**
Note that CommonLogHolder.setK8sInfo must be called somewhere because currently Logback integration is resolutely not spring compatible.

Can be called in ServiceLoggingConfiguration, where CommonLogHolder.setServiceType is also called.

@seif this is the first implementation unless you've beaten me to it?
@fessyfoo fyi

### CODE REVIEW

**Comment by mikebell90** [comment]
Requires
* otj-spring 1.3.11
* otj-server 3.0.23

File: core/src/test/java/com/opentable/logging/JsonLogEncoderTest.java [hunk 1]
```diff
@@ -23,18 +25,47 @@
...
+
+    @Test
+    public void encodeWithMockKubernetes() throws Exception {
+       CommonLogHolder.setK8sInfo(getMockK8sInfo());
+       final ObjectNode node = commonTests();
+       assertEquals("logging-cluster-name", node.get("k8s-cluster-name").asText());
+        assertEquals("logging-namespace", node.get("k8s-namespace").asText());
+        assertEquals("logging-node-host", node.get("k8s-node-host").asText());
+        assertEquals("logging-servicename", node.get("k8s-service-name").asText());
+        assertEquals("logging-podname", node.get("k8s-pod-name").asText());
```
Comment (rushalias) [inline core/src/test/java/com/opentable/logging/JsonLogEncoderTest.java]: 👍

File: core/src/main/java/com/opentable/logging/CommonLogFields.java [hunk 2]
```diff
@@ -171,4 +171,40 @@ default long getSequenceNumber() {
...
+     * @return the cluster name
+     */
+    @JsonProperty("k8s-cluster-name")
+    String getClusterName();
+
+    /**
+     * Kubernetes namespace
+     * @return the namespace
+     */
+    @JsonProperty("k8s-namespace")
```
Comment (adamzr) [inline core/src/main/java/com/opentable/logging/CommonLogFields.java]: Are these all in the OTL?

Comment (mikebell90) [reply core/src/main/java/com/opentable/logging/CommonLogFields.java]: yes

**Pull request merged**